data_model/src/permission.rs

```diff
@@ -1,212 +1,124 @@
 #[cfg(not(feature = "std"))]
-use alloc::{borrow::ToOwned as _, collections::BTreeSet, format, string::String, vec::Vec};
+use alloc::{collections::BTreeSet, format, string::String, vec::Vec};
+use core::borrow::Borrow;
 #[cfg(feature = "std")]
 use std::collections::BTreeSet;
 
-use iroha_data_model_derive::model;
+use iroha_data_model_derive::{model, IdEqOrdHash};
 use iroha_schema::IntoSchema;
 use parity_scale_codec::{Decode, Encode};
 use serde::{Deserialize, Serialize};
 
 pub use self::model::*;
-use crate::name::Name;
+use crate::{executor::ExecutorDataModelObject, name::Name};
 
 /// Collection of [`Token`]s
 pub type Permissions = BTreeSet<PermissionToken>;
 
 use super::*;
 
-/// Unique id of [`PermissionToken`]
-pub type PermissionTokenId = Name;
-
 #[model]
 mod model {
     use super::*;
 
-    /// Stored proof of the account having a permission for a certain action.
-    ///
-    /// Since permission token is represented opaque to core
-    /// either executor or client should make sure that tokens are represented uniformly.
-    ///
-    /// So that:
-    /// - payload A is equal to payload B then token A must be equal to token B
-    /// - and if payload A is't equal to B then token A mustn't be equal to token B
+    /// Identification of [`PermissionToken`].
     #[derive(
         Debug,
+        Display,
         Clone,
         PartialEq,
         Eq,
         PartialOrd,
         Ord,
+        Hash,
+        Constructor,
+        FromStr,
+        Getters,
         Decode,
         Encode,
         Deserialize,
         Serialize,
         IntoSchema,
     )]
-    #[ffi_type]
-    pub struct PermissionToken {
-        /// Token identifier
-        pub definition_id: PermissionTokenId,
-        /// JSON encoded token payload
-        pub payload: JsonString,
+    #[getset(get = "pub")]
+    #[serde(transparent)]
+    #[repr(transparent)]
+    #[ffi_type(opaque)]
+    pub struct PermissionTokenId {
+        /// Should be unique.
+        pub name: Name,
     }
 
-    /// Description of tokens defined in the executor
+    /// Stored proof of the account having a permission for a certain action.
     #[derive(
-        Debug,
-        Display,
-        Clone,
-        PartialEq,
-        Eq,
-        PartialOrd,
-        Ord,
-        Default,
-        Decode,
-        Encode,
-        Deserialize,
-        Serialize,
-        IntoSchema,
+        Debug, Clone, IdEqOrdHash, Decode, Encode, Deserialize, Serialize, IntoSchema, Display,
     )]
-    #[display(fmt = "{token_ids:#?}")]
     #[ffi_type]
-    pub struct PermissionTokenSchema {
-        /// Ids of all permission tokens, sorted.
-        pub token_ids: Vec<PermissionTokenId>,
-        /// Type schema of permission tokens
+    #[display(fmt = "PERMISSION TOKEN `{definition_id}`")]
+    pub struct PermissionToken {
+        /// Refers to a type defined in [`crate::executor::ExecutorDataModel`].
+        #[id]
+        pub definition_id: PermissionTokenId,
```

Comment:
```suggestion
        pub definition_id: PermissionTokenId,
```

you cannot compare permission tokens only with their id because 2 permission tokens can have the same ID but different payloads. Example:
`CanUnregisterDomain {domain_id: "wonderland"}`  != `CanUnregisterDomain {domain_id: "noland"}`

Reply:
So, the primary purpose of `#[id]` is to quickly compare of equality?

Does this mean that `PermissionToken` & `Parameter` cannot implement `Identifiable`?

Reply:
> Does this mean that `PermissionToken` & `Parameter` cannot implement Identifiable?

yes, it means that `PermissionToken` cannot implement `Identifiable`
no, it doesn't mean that `Parameter` cannot implement `Identifiable`

Reply:
> yes, it means that `PermissionToken` cannot implement Identifiable

user can have assigned multiple (as I pointed out in the example) permissions of the same id with different payloads.

>  no, it doesn't mean that Parameter cannot implement Identifiable

`ParameterId` uniquely identifies a parameter, it's not contingent on the payload